lib/Common/Codex/Utf8Codex.cpp

```diff
@@ -395,10 +394,11 @@ namespace utf8
             cch -= 4;
         }
 LSlowPath:
+        // WARNING: Assume cch correct, suppress end-of-buffer checking
+        // we might have failed to measure cch, +1 for odd length. sizeof(char16) == 2
+        LPCUTF8 end = ptr + cch + (cch % 2 ? 2 : 1);
         while (cch-- > 0)
         {
-            LPCUTF8 end = ptr + cch + 1; // WARNING: Assume cch correct, suppress end-of-buffer checking
```

Comment:
> The original one was "*buffer++ = Decode(ptr, ptr + 4, localOptions)" and not broken... except that it ran into issues in some boundary case that Taylor had to fix.

It is/was broken.

Lets keep the boundary cases, and fastPath problems aside.. `EncodedBytes` may perfectly return <1 or >4 and when it does...